cloudinit/net/dhcp.py

```diff
@@ -107,6 +110,26 @@ def obtain_lease(self):
         self._ephipv4 = ephipv4
         return self.lease
 
+    def extract_dhcp_options_mapping(self, nmap):
+        result = {}
+        for internal_reference, lease_option_names in nmap.items():
+            if isinstance(lease_option_names, list):
+                self.get_first_option_value(
+                    internal_reference,
+                    lease_option_names,
+                    result
+                )
+            else:
+                result[internal_reference] = self.lease.get(lease_option_names)
+        return result
+
+    def get_first_option_value(
```

Comment:
```suggestion
    def get_first_option_value(self, internal_mapping,
                               lease_option_names, result):
```
Let's just wrap the parameters so we don't have a ): at the start of the line.

Reply:
> A couple more changes please. Thanks for working to get this fix included.

I'm doing my best to support. :) I really think this is a step in the right direction.  

And please don't worry about me; I need to learn and I think we need to share our experiences.  No one knows more than another.  If you disagree or aren't comfortable with some part of the code, I encourage you to say so :).  I will do the same, but all in the name of making it more readable/understandable by others.